src/redux/trainers/thunks.js

```diff
@@ -5,8 +5,82 @@ export const getTrainers = async (dispatch) => {
     dispatch(trainersActions.getTrainersPending());
     const response = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers`);
     const data = await response.json();
-    dispatch(trainersActions.getTrainersSuccess(data.data));
+    if (!data.error) {
+      dispatch(trainersActions.getTrainersSuccess(data.data));
+    }
+    if (data.error) {
+      throw new Error(data.message);
+    }
   } catch (error) {
-    dispatch(trainersActions.getTrainersError(error.toString()));
+    dispatch(trainersActions.getTrainersError(error.message));
+  }
+};
+export const deleteTrainer = async (dispatch, id) => {
+  dispatch(trainersActions.deleteTrainersPending());
+  try {
+    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers/${id}`, {
+      method: 'DELETE'
+    });
+    const { message, error } = await response.json();
+    if (!error) {
+      dispatch(trainersActions.deleteTrainersSuccess(id, message));
+    } else {
+      throw new Error(message);
+    }
+  } catch (error) {
+    dispatch(trainersActions.deleteTrainersError(error.toString()));
+  }
+};
+export const sendTrainer = async (dispatch, item) => {
+  try {
+    dispatch(trainersActions.addTrainersPending());
+    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers/`, {
+      method: 'POST',
+      body: JSON.stringify(item),
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    });
+    const { data, message, error } = await response.json();
+    if (!error) {
+      dispatch(trainersActions.addTrainersSuccess(data, message));
+    }
+    if (error) {
+      throw new Error(message);
+    }
+  } catch (error) {
+    dispatch(trainersActions.addTrainersError(error.message));
+  }
+};
+export const putTrainer = async (dispatch, id, updatedTrainer) => {
+  const trainer = {
+    firstName: updatedTrainer.firstName,
+    lastName: updatedTrainer.lastName,
+    dni: updatedTrainer.dni.toString(),
+    phone: updatedTrainer.phone.toString(),
+    email: updatedTrainer.email,
+    city: updatedTrainer.city,
+    password: updatedTrainer.password,
+    salary: updatedTrainer.salary.toString()
+  };
+  try {
+    dispatch(trainersActions.editTrainersPending());
+    console.log(updatedTrainer);
+    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers/${id}`, {
+      method: 'PUT',
+      body: JSON.stringify(trainer),
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    });
+    const { data, message, error } = await response.json();
+    if (!error) {
+      dispatch(trainersActions.editTrainersSuccess(data.data, id, message));
+    }
```

Comment:
The error that santi said it's because of this, should be data not data.data because you have it already deconstructured

Reply:
If you change this everything should work